presto/presto.py

```diff
@@ -458,12 +486,19 @@ def forward(
 
         # append latlon tokens
         latlon_tokens = self.latlon_embed(self.cartesian(latlons)).unsqueeze(1)
-        x = torch.cat((latlon_tokens, x), dim=1)
-        upd_mask = torch.cat((torch.zeros(x.shape[0])[:, None].to(device), upd_mask), dim=1)
-        orig_indices = torch.cat(
-            (torch.zeros(x.shape[0])[:, None].to(device).int(), orig_indices + 1),
-            dim=1,
-        )
+        x, upd_mask, orig_indices = self.add_token(latlon_tokens, x, upd_mask, orig_indices)
```

Comment:
@cbutsko a hackey way to get the model to ignore latlons is to change the mask here. A mask value of 0s tells the model to include the value, and a mask value of 1s tells it to ignore the value.

New tokens get added to the front of the sequence. Concretely:

`x` has shape `[batch_size, num_tokens, dim]`
and `mask` has shape `[batch_size, num_tokens]`.

The latlons just got added to the front of the sequence, so you can do

```python
upd_mask[:, 0] = 1
```
right after line 489 (where `self.add_token(latlon_tokens, x, upd_mask, orig_indices)` is called) to update the mask so that the latlon token will be ignored entirely.

Reply:
added this line as you suggested instead of filling latlons with zeros in the dataset
https://github.com/WorldCereal/presto-worldcereal/commit/0cf1e6072e129499fcc205ee11d3b7dc43dad472